ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -582,7 +606,10 @@ static KsqlRestApplication buildApplication(
         Duration.ofMillis(restConfig.getLong(DISTRIBUTED_COMMAND_RESPONSE_TIMEOUT_MS_CONFIG)),
         versionChecker::updateLastRequestTime,
         authorizationValidator,
-        errorHandler
+        errorHandler,
+        restConfig.getBoolean(KsqlRestConfig.KSQL_QUERY_STANDBY_ENABLE_CONFIG),
+        restConfig.getBoolean(KsqlRestConfig.KSQL_QUERY_STANDBY_ENABLE_CONFIG),
+        hostStatusEntityMap
```

Comment:
Passing an immutable map through the constructor doesn't allow you to change the hosts.  Also, if you assume that the HostStatusEntity objects will have their active field set dynamically, that's not threadsafe.  If you notice in my change, I make it immutable.

You probably want to pass a supplier which will call `getHostsStatus` every time rather than pass the map itself.

Reply:
This code moved into the `LivenessFilter` where the get is called every time we need to access the map. Regarding thread safety, the code accessed an immutable map hence the data cannot get corrupted. correct?